### PR TITLE
✨ RedisPipeline

### PR DESCRIPTION
**Pull request by RTLS**

### CODE REVIEW

**Comment by RTLS** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/556?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#556** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/556?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/556?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#555** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/555?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lib/sequin/redis/redis.ex [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  @callback send_messages(%RedisDestination{}, [any()]) :: :ok | {:error, Error.t()}
+
+  @spec send_messages(%RedisDestination{}, [any()]) :: :ok | {:error, Error.t()}
+  def send_messages(%RedisDestination{} = destination, messages) do
+    impl().send_messages(destination, messages)
+  end
+
+  defp impl do
+    Application.get_env(:sequin, :redis_module, Sequin.Redis.Client)
+  end
```
Comment (acco) [inline lib/sequin/redis/redis.ex:15]: 🚀

File: test/sequin/redis_pipeline_test.exs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  alias Sequin.Factory.AccountsFactory
+  alias Sequin.Factory.CharacterFactory
+  alias Sequin.Factory.ConsumersFactory
+  alias Sequin.Factory.DatabasesFactory
+  alias Sequin.Factory.ReplicationFactory
+  alias Sequin.Test.Support.Models.Character
+  alias Sequin.Test.Support.Models.CharacterDetailed
+
+  # Make sure mocks are verified when the test exits
+  setup :verify_on_exit!
```
Comment (acco) [inline test/sequin/redis_pipeline_test.exs:18]: We do this globally: https://github.com/sequinstream/sequin/blob/master/test/support/case.ex#L35

👍

File: test/sequin/redis_pipeline_test.exs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          sequence_filter: ConsumersFactory.sequence_filter_attrs(group_column_attnums: [1]),
+          sequence_id: sequence.id,
+          replication_slot_id: replication.id
+        )
+
+      {:ok, %{consumer: consumer}}
+    end
+
+    test "events are sent to Redis", %{consumer: consumer} do
+      # test_pid = self()
```
Comment (acco) [inline test/sequin/redis_pipeline_test.exs:55]: del

File: test/sequin/redis_pipeline_test.exs [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+          account_id: account.id,
+          type: :redis,
+          replication_slot_id: replication.id,
+          sequence_id: sequence.id,
+          sequence: sequence,
+          message_kind: :record,
+          postgres_database: database
+        )
+
+      :ok = Consumers.create_consumer_partition(consumer)
```
Comment (acco) [inline test/sequin/redis_pipeline_test.exs:145]: Thank god you figured it out! 🙏

**Review by acco — APPROVED** [review]
SO good!

**Pull request merged**